app/models/concerns/attachment_jcamp_aasm.rb

```diff
@@ -350,6 +349,21 @@ def delete_edit_peak_after_done
     destroy if %w[edit peak].include?(typname)
   end
 
+  def delete_related_edit_peak(jcamp_att)
+    return unless jcamp_att
+
+    atts = Attachment.where(attachable_id: attachable_id)
+    valid_name = fname_wo_ext(self)
+    atts.each do |att|
+      is_delete = (
+        (att.edited? || att.peaked?) &&
+          att.id != jcamp_att.id &&
+          valid_name == fname_wo_ext(att)
+      )
+      att.delete if is_delete
+    end
+  end
+
   def fname_wo_ext(target)
     parts = target.filename_parts
     ending = parts.length == 2 || parts.length == 3 ? -2 : -3
```

Comment:
What if the original filename has more than 2 dots? 
Constructed case: 
 **x1.x2.x3.jdx** would lead to a file_name x1_x2
 **x1.x2.y3.jdx** would also lead to a file_name x1_x2